tests/unit/stacked-vertical-bar-test.js

```diff
@@ -0,0 +1,773 @@
+  var dataByBarLabel, grossBarSums, minGrossBarSum, expectedBarHeightRatios,
+    actualBarHeights, minActualBarHeight, barLabel, expectedBarHeight;
+
+  dataByBarLabel = _.groupBy(three_ranges, 'barLabel');
+  assert.expect(_.keys(dataByBarLabel).length);
+
+  // Compute the expected heights of each bar as a percentage
+  // of the expected height of the shortest bar.
+  grossBarSums = _.mapValues(dataByBarLabel, function(barData) {
+    var sum = 0.0;
+    for (var iDatum = 0; iDatum < barData.length; iDatum++) {
+      sum += Math.abs(barData[iDatum].value);
+    }
+    return sum;
+  });
+  minGrossBarSum = _.min(_.values(grossBarSums));
+  expectedBarHeightRatios = _.mapValues(grossBarSums, function(sum) {
+    return (sum / minGrossBarSum);
+  });
+
+  this.subject({data: three_ranges});
+  this.render();
+
+  // Find the height of each SVG bar by summing the heights of its slices,
+  // as well as the shortest height.
+  actualBarHeights = {};
+  this.$('svg g.bars').each(function() {
+    var slices = $('rect', this);
+    actualBarHeights[$(this).text()] = _.sum(_.map(slices, getHeightOfSvgSlice));
+  });
+  minActualBarHeight = _.min(_.values(actualBarHeights));
+
+  // Compare the expected and actual heights
+  // and assert that they are approximately equal.
+  for (barLabel in expectedBarHeightRatios) {
+    expectedBarHeight = minActualBarHeight * expectedBarHeightRatios[barLabel];
+    assert.ok(
+      equalsWithTolerance(
+        actualBarHeights[barLabel],
+        expectedBarHeight),
+      "The bar for '" + barLabel +
+        "' has a height of " + expectedBarHeight +
+        " px (actual height: " + actualBarHeights[barLabel] +
+        " px)");
+  }
+});
+
+test('The bars have the correct heights relative to the values on the y-axis ticks', function(assert) {
+  var dataByBarLabel, grossBarSums, pixelsPerDataUnit, expectedBarHeights,
+    actualBarHeights, barLabel;
+
+  dataByBarLabel = _.groupBy(three_ranges, 'barLabel');
+  assert.expect(1 + _.keys(dataByBarLabel).length);
+
+  // Compute the expected heights of each bar as a percentage
+  // of the expected height of the shortest bar.
+  grossBarSums = _.mapValues(dataByBarLabel, function(barData) {
+    var sum = 0.0;
+    for (var iDatum = 0; iDatum < barData.length; iDatum++) {
+      sum += Math.abs(barData[iDatum].value);
+    }
+    return sum;
+  });
+
+  this.subject({data: three_ranges});
+  this.render();
+
+  // Find the number of pixels between consecutive y-axis ticks,
+  // and the data value difference that number of pixels represents.
+  pixelsPerDataUnit = (() => {
+    assert.ok(this.$('svg g.y.axis g.tick').length >= 2,
+      "There are at least 2 Y-axis ticks in the graph");
+
+    var firstTickSvg = this.$('svg g.y.axis g.tick')[0];
+    var secondTickSvg = this.$('svg g.y.axis g.tick')[1];
+
+    var firstTickTransformY =
+      parseFloat($(firstTickSvg).attr('transform').match(/\d+(\.\d+)?/g)[1]);
```

Comment:
Each Y-axis tick element looks like this:

    <g class="tick" transform="translate(0.0, 42.0)" ... />

The regex is supposed to extract the y-delta, that is, the `42.0` above.

I will rewrite this without using a regex, but I will warn you that the result will be no clearer, and will still need a long comment in the code to explain. Hint:

    var firstTickTransformY = firstTickSvg.transform.baseVal[0].matrix.f;